src/screens/Assistant_v2/use-trips-query.ts

```diff
@@ -225,3 +232,45 @@ const stringifyLocation = (location: Location | undefined) => {
   if (!location) return 'Undefined location';
   return `${location.id}--${location.name}--${location.locality}`;
 };
+
+function generateKeyFromTripPattern(tripPattern: TripPattern) {
+  const firstServiceLeg = tripPattern.legs.find((leg) => leg.serviceJourney);
+  const key =
+    (firstServiceLeg?.aimedStartTime ?? '') +
+    tripPattern.legs
+      .map((leg) => {
+        return leg.toPlace.longitude.toString() + leg.toPlace.latitude;
+      })
+      .join('-');
+
+  return key;
+}
+
+function decorateTripPatternWithKey(
+  tripPatterns: TripPattern[],
+): TripPatternWithKey[] {
+  return tripPatterns.map((tp) => {
+    (tp as TripPatternWithKey).key = generateKeyFromTripPattern(tp);
+    return tp as TripPatternWithKey;
```

Comment:
Akkurat i denne bruken vil dette gå fint for tripPatterns som sendes inn er akkurat generert og hentet fra server. men verdt å merke seg at om man feks vil bruke decorateTripPatternWithKey på data som er f.eks i state eller blir hentet ut fra en annen plass så kan det gi uforutsigbare virkninger igjennom mutasjonen.

```suggestion
    return {
      ...tp,
      key: generateKeyFromTripPattern(tp)
    };
```

unngår vel også noe casting

Reply:
decorateWithKey exporteres ikke ut fra trip-query-hooken, og kan dermed i utgangspunktet ikke brukes utenfor hooken.
Inne i hooken har vi full kontroll på datatypen vi bruker, og her sørges det for at hooken alltid leverer TripPatternWithKey.
Typen til inn-parameteren er også eksplisitt hentet fra TripsQuery. 
Det bør ringe noen bjeller hvis noen tenker å exportere decorate og bruke den på random TripPatterns.
endringsforslaget er bra, liker at vi unngår casting.